src/components/fields/AnyOfField.js

```diff
@@ -0,0 +1,147 @@
+              delete newFormData[key];
+            }
+          }
+        }
+      }
+
+      onChange(newFormData);
+    } else {
+      onChange(undefined);
+    }
+
+    this.setState({
+      selectedOption: parseInt(event.target.value, 10),
+    });
+  };
+
+  render() {
+    const {
+      disabled,
+      errorSchema,
+      formData,
+      idPrefix,
+      idSchema,
+      onBlur,
+      onChange,
+      onFocus,
+      schema,
+      registry,
+      safeRenderCompletion,
+      uiSchema,
+    } = this.props;
+    const _SchemaField = registry.fields.SchemaField;
+    const { selectedOption } = this.state;
+
+    const baseType = schema.type;
+    const options = schema.anyOf || [];
+
+    return (
+      <div className="panel panel-default panel-body">
+        <div className="form-group">
+          <select
+            className="form-control"
+            onChange={this.onOptionChange}
+            value={selectedOption}
+            id={`${idSchema.$id}_anyof_select`}>
+            {options.map((option, index) => {
+              return (
+                <option key={index} value={index}>
+                  {option.title || `Option ${index + 1}`}
+                </option>
+              );
+            })}
+          </select>
+        </div>
+
+        {options.map((option, index) => {
+          if (index !== selectedOption) {
+            return null;
+          }
+
+          // If the subschema doesn't declare a type, infer the type from the
+          // parent schema
+          const optionSchema = option.type
+            ? option
+            : Object.assign({}, option, { type: baseType });
+
+          return (
+            <_SchemaField
+              key={index}
+              schema={optionSchema}
+              uiSchema={uiSchema}
+              errorSchema={errorSchema}
+              idSchema={idSchema}
+              idPrefix={idPrefix}
+              formData={formData}
+              onChange={onChange}
+              onBlur={onBlur}
+              onFocus={onFocus}
+              registry={registry}
+              safeRenderCompletion={safeRenderCompletion}
+              disabled={disabled}
+            />
+          );
+        })}
+      </div>
+    );
+  }
+}
+
+AnyOfField.defaultProps = {
+  disabled: false,
+  errorSchema: {},
+  idSchema: {},
+  uiSchema: {},
+};
+
+if (process.env.NODE_ENV !== "production") {
+  AnyOfField.propTypes = {
+    schema: PropTypes.object.isRequired,
+    uiSchema: PropTypes.object,
+    idSchema: PropTypes.object,
+    formData: PropTypes.any,
+    errorSchema: PropTypes.object,
+    registry: PropTypes.shape({
+      widgets: PropTypes.objectOf(
+        PropTypes.oneOfType([PropTypes.func, PropTypes.object])
+      ).isRequired,
+      fields: PropTypes.objectOf(PropTypes.func).isRequired,
+      definitions: PropTypes.object.isRequired,
+      ArrayFieldTemplate: PropTypes.func,
+      ObjectFieldTemplate: PropTypes.func,
```

Comment:
Why do we need `ArrayFieldTemplate` or `ObjectFieldTemplate`?

Reply:
Although, looking at this code further, is this not just describing the shape of the registry object?

Reply:
To be honest, I'm not a React expert so I'm not really sure myself. It feels silly for us to have to redescribe the registry object's shape in every single component. If you want to leave it as-is for now and file a bug to clean it up later, that's fine.

Reply:
@glasserc I think we can use `PropTypes.shape()` to create a re-usable definition for the registry